### ISSUE TITLE
A search hangs and doesn't return

### ISSUE THREAD

**Issue by Sembiance** [typesense/typesense#789]
When running typesense `0.24.0.rcn25-log1` I've got a particular search query that just runs forever.

Exact reproducible steps below, including DB data.

Download the data file (13GB) and extract (28GB) and start up typesense (27GB RAM):
```
wget https://telparia.com/wip/hangingSearch.tar.bz2"
tar -xvf hangingSearch.tar.gz
cd hangingSearch
typesense-server --data-dir=./data --api-key=TESTCASEKEY --api-address=127.0.0.1 --api-port=8108 --peering-address=127.0.0.1 --peering-port=8107
```

Wait for the DB to load, then perform the search query:
```
curl --silent "http://127.0.0.1:8108/multi_search?query_by=content,ext,filename,title" -X POST -H "Content-Type: application/json" -H "X-TYPESENSE-API-KEY: TESTCASEKEY" --data-binary @"hangingSearch.json"
```

The search will run forever, or seemingly forever.

If this is not a bug, then it's important that `search_cutoff_ms` be enhanced so that is capable of stopping these long running searches, otherwise the DB won't ever be able to be written to because the search will be locking it up.

**Comment by Sembiance** [comment]
So I have a theory about this.

The schema was created with a `token_seperators` value that includes period `.` as a sperator. The stuck search query is `Windows 3.1` which which would then be resolved as `Windows` `3` `1` which I imagine the DB is getting hung up on the fact that there are so many matches for single digits.

I have since encountered 2 other searches that hang (with full dataset, not sure if it hangs with the provided subset of data): `bbs lord 4.0` and `sam 4.5.1`. Both of these would result in single digit terms after the token split.

I'm guessing this is what is nagung up typesense. I still think it's important that `search_cutoff_ms` can interrupt these searches, but in the mean time maybe I'll filter out single digit tokens before sending to typesense.

**Comment by kishorenc** [comment]
Fixed in `v0.24`
